sql: TestIndexBackfillAfterGC failed

sql.TestIndexBackfillAfterGC [failed](https://mesolite.cluster.engflow.com/invocations/default/0a975efa-cc51-4c0a-860c-d937b79d242b?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on release-26.1 @ [2dede1c13f1bf518511d790900cbd47e0557472f](https://github.com/cockroachdb/cockroach/commits/2dede1c13f1bf518511d790900cbd47e0557472f):


```
=== RUN   TestIndexBackfillAfterGC
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestIndexBackfillAfterGC845146453
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:668: DRPC is enabled (via override by TestingGlobalDRPCOption)
    schema_changer_test.go:4228: -- test log scope end --
test logs left over in: outputs.zip/logTestIndexBackfillAfterGC845146453
--- FAIL: TestIndexBackfillAfterGC (78.95s)
=== RUN   TestIndexBackfillAfterGC/useDeclarative=false
    schema_changer_test.go:4210: pq: polling for queued jobs to complete: poll-show-jobs: batch timestamp 1769331990.714233634,0 must be after replica GC threshold 1769331990.715164707,0 (r81: /Tenant/10{-/Table/106/2})
--- FAIL: TestIndexBackfillAfterGC/useDeclarative=false (0.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #161153 sql: TestIndexBackfillAfterGC failed [C-test-failure O-robot P-3 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexBackfillAfterGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59063

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -4149,6 +4149,7 @@ func TestSchemaChangeErrorOnCommit(t *testing.T) {
 func TestIndexBackfillAfterGC(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderDuress(t, "flaky under race")
 
 	var tc serverutils.TestClusterInterface
 	ctx := context.Background()

